s3/src/test/java/ch/cyberduck/core/cryptomator/UVFTest.java

```diff
@@ -0,0 +1,196 @@
+                "    \"kdf\": \"HKDF-SHA512\",\n" +
+                "    \"kdfSalt\": \"NIlr89R7FhochyP4yuXZmDqCnQ0dBB3UZ2D+6oiIjr8=\",\n" +
+                "    \"org.example.customfield\": 42\n" +
+                "}";
+
+        try {
+            for(final String fi : files) {
+                final Path file = new Path("/" + bucketName + "/" + fi, EnumSet.of(AbstractPath.Type.file));
+                byte[] content = new byte[1000];
+                final int size = UVFTest.class.getResourceAsStream("/uvf/first_vault" + fi).read(content);
+                final TransferStatus transferStatus = new TransferStatus().withLength(size);
+                transferStatus.setChecksum(storage.getFeature(Write.class).checksum(file, transferStatus).compute(new ByteArrayInputStream(content), transferStatus));
+                storage.getFeature(Bulk.class).pre(Transfer.Type.upload, Collections.singletonMap(new TransferItem(file), transferStatus), new DisabledConnectionCallback());
+                final StatusOutputStream<?> out = storage.getFeature(Write.class).write(file, transferStatus, new DisabledConnectionCallback());
+                IOUtils.copyLarge(UVFTest.class.getResourceAsStream("/uvf/first_vault" + fi), out);
+                out.close();
+            }
+
+            final VaultRegistry vaults = new DefaultVaultRegistry(new DisabledPasswordCallback());
+            bookmark.setDefaultPath("/" + bucketName);
+            final UVFVault vault = new UVFVault(new DefaultPathHomeFeature(bookmark).find());
+            vaults.add(vault.load(storage, new DisabledPasswordCallback() {
+                @Override
+                public Credentials prompt(final Host bookmark, final String title, final String reason, final LoginOptions options) {
+                    return new Credentials().withPassword(jwe);
+                }
+            }));
+            final PathAttributes attr = storage.getFeature(AttributesFinder.class).find(vault.getHome());
+            try(
+                    UVFMasterkey masterKey = UVFMasterkey.fromDecryptedPayload(jwe)) {
+                attr.setDirectoryId(masterKey.rootDirId());
+            }
+            storage.withRegistry(vaults);
+
+            final Path home = vault.getHome().withAttributes(attr).withType(EnumSet.of(AbstractPath.Type.directory, AbstractPath.Type.vault));
+            {
+                final AttributedList<Path> list = storage.getFeature(ListService.class).list(home, new DisabledListProgressListener());
+                assertEquals(2, list.size());
+                assertTrue(Arrays.toString(list.toArray()), list.contains(new Path("/cyberduckbucket/foo.txt", EnumSet.of(AbstractPath.Type.file, AbstractPath.Type.decrypted))));
+                assertTrue(Arrays.toString(list.toArray()), list.contains(new Path("/cyberduckbucket/subdir", EnumSet.of(AbstractPath.Type.directory, AbstractPath.Type.placeholder, AbstractPath.Type.decrypted))));
```

Comment:
@dkocher @ylangisc the path comes without directory ID; when listing the subdir below, the call fails when encrytping the plain name. Is something missing in the UVF implementation or is there something missing in my test?

Reply:
In 01539ae.